cdap-ui/app/cdap/components/FileBrowser/index.js

```diff
@@ -0,0 +1,306 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import MyDataPrepApi from 'api/dataprep';
+import NamespaceStore from 'services/NamespaceStore';
+import shortid from 'shortid';
+import classnames from 'classnames';
+import moment from 'moment';
+import {Link} from 'react-router';
+import FilePath from 'components/FileBrowser/FilePath';
+import {convertBytesToHumanReadable, HUMANREADABLESTORAGE_NODECIMAL} from 'services/helpers';
+import T from 'i18n-react';
+
+require('./FileBrowser.scss');
+
+const BASEPATH = '/Users';
+const PREFIX = 'features.FileBrowser';
+
+export default class FileBrowser extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      contents: [],
+      path: '',
+      statePath: this.props.pathname,
+      error: null,
+      loading: true,
+      search: ''
+    };
+
+    this.handleSearch = this.handleSearch.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.fetchDirectory(nextProps);
+  }
+
+  fetchDirectory(props) {
+    let hdfsPath;
+
+    if (this.props.noState) {
+      hdfsPath = this.props.initialDirectoryPath;
+      return;
+    } else {
+      hdfsPath = props.location.pathname.slice(props.pathname.length);
+      hdfsPath = hdfsPath || this.props.initialDirectoryPath || BASEPATH;
+    }
+
+    if (hdfsPath === this.state.path) { return; }
+
+    this.goToPath(hdfsPath);
+  }
+
+  goToPath(path) {
+    this.setState({loading: true});
+
+    let namespace = NamespaceStore.getState().selectedNamespace;
+
+    MyDataPrepApi.explorer({
+      namespace,
+      path
+    }).subscribe((res) => {
+      this.setState({
+        loading: false,
+        contents: this.formatResponse(res.values),
+        error: null,
+        search: '',
+        path
+      });
+    }, (err) => {
+      this.setState({
+        loading: false,
+        error: err.response,
+        search: '',
+        path
+      });
+    });
+  }
+
+  formatResponse(contents) {
+    return contents.map((content) => {
+      content.uniqueId = shortid.generate();
+      content['last-modified'] = moment(content['last-modified']).format('MM/DD/YY HH:mm');
+      content.size = convertBytesToHumanReadable(content.size, HUMANREADABLESTORAGE_NODECIMAL);
+      content.type = content.type === 'UNKNOWN' ? '--' : content.type;
+      return content;
+    });
+  }
+
+  handleSearch(e) {
+    this.setState({search: e.target.value});
+  }
+
+  renderRow(row) {
+    return (
+      <div
+        key={row.uniqueId}
+        className="row content-row"
+      >
+        <div className="col-xs-3 name">
+          <span
```

Comment:
I would suggest we move away from this pattern. We would lose keyboard accessibility if we do this. We should change this in dataprep as well.